internal/storage/postgre.go

```diff
@@ -23,35 +27,161 @@ func NewPostgreBase(db *pgx.Conn, log *zap.Logger) *Postgre {
 	}
 }
 
+func Migrations(dsn string) error {
+	db, err := sql.Open("pgx", dsn)
+	if err != nil {
+		return err
+	}
+	defer db.Close()
+
+	if err := goose.SetDialect("postgres"); err != nil {
+		return err
+	}
+	if err := goose.Up(db, "internal/db/migrations"); err != nil {
+		return err
+	}
+	return nil
+}
+
 func (p *Postgre) Ping() error {
 	p.Log.Debug("Проверяем статус соединения с БД")
 	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
 	defer cancel()
 	if err := p.db.Ping(ctx); err != nil {
-		p.Log.Debug("Ошибка соединения с БД")
+		p.Log.Error("Ошибка соединения с БД")
 		fmt.Println(err)
 		return err
 	}
-	p.Log.Debug("Есть соединение с БД")
+	p.Log.Info("Есть соединение с БД")
 	return nil
 }
 
-func (p *Postgre) AddURL(data URLData) error {
-	//TODO implement me
-	panic("implement me1")
+func (p *Postgre) AddURL(ctx context.Context, data URLData) error {
+	p.Log.Debug("Добавляем URL в базу данных", zap.String("url", data.URL))
+	_, err := p.db.Exec(ctx, "INSERT INTO urls (id, url) VALUES ($1, $2)", data.ID, data.URL)
+	if err != nil {
+		return err
+	}
+	return nil
 }
 
-func (p *Postgre) CheckID(id string) (URLData, bool, error) {
-	//TODO implement me
-	panic("implement me2")
+func (p *Postgre) AddBatchURL(ctx context.Context, urls map[string]ReqBatchJSON) error {
+	tx, err := p.db.Begin(ctx)
+	if err != nil {
+		return err
+	}
+	for id, v := range urls {
+		// все изменения записываются в транзакцию
+		if v.Valid && !v.Exist {
+			_, err := tx.Exec(ctx,
```

Comment:
CHANGE REQUEST: please optimize it

https://stackoverflow.com/questions/70489530/golang-route-for-insert-preparecontext-error